transport.go

```diff
@@ -17,26 +21,43 @@ type Transport struct {
 	Base http.RoundTripper
 	// source supplies the token to use when signing a request
 	source TokenSource
-	// auther adds OAuth1 Authorization headers to requests
-	auther *auther
+	// Auther adds OAuth1 Authorization headers to requests
+	auther Auther
+}
+
+// NewTransport returns a new Transport and an error
+func NewTransport(baseRoundTripper http.RoundTripper, source TokenSource, auther Auther) (*Transport, error) {
+	t := newTransport(baseRoundTripper, source, auther)
+	err := t.checkValid()
+	if err != nil {
+		return nil, err
+	}
+	return t, nil
+}
+
+// newTransport returns a new Transport without checking whether there is an error
```

Comment:
is there any circumstances we don't want to check parameters beforehand ?

Reply:
because there is no space to return error in existing [NewClient](https://github.com/rayark/oauth1/blob/ac254403590145a865f2f9fb213a59e07c5b7c5d/config.go#L52)

Reply:
maybe leave a comment to make it clear that `newTransport` is only for `NewTransport` & `NewClient`

Reply:
fixed in 0f1df77259de5323282abb51f6162a3bd3fb7905